src/components/List/ListItem/ListItem.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import Highlighter from 'react-highlight-words';
+import { TestProps } from 'utils/types';
+
+import { ListItemType, ListRowSize, SelectHandlerType } from '../types';
+import { listItemStyle, contentStyle } from './ListItem.style';
+
+type Props = {
+  /** Size of the ListItem (translates to height) */
+  size: ListRowSize;
+  /** Content of the ListItem */
+  content: ListItemType;
+  /** Index, for test-id calculation */
+  index: number;
+  /** Selected state */
+  selected?: boolean;
+  /** Whether the text of the ListItem is highlighted or not. eg: Filter - Default Value */
+  highlighted?: boolean;
+  /** Disabled state */
+  disabled?: boolean;
+  /** Search Term to be highlighted in list items */
+  searchTerm?: string;
+  /** Option Click handler for SelectOption[] data case */
+  handleOptionClick?: SelectHandlerType;
+} & TestProps;
+
+const ListItem = React.forwardRef<HTMLDivElement, Props>(
+  (
+    {
+      size,
+      content,
+      index,
+      selected = false,
+      highlighted = false,
+      disabled = false,
+      handleOptionClick,
+      searchTerm,
+      dataTestId,
+    },
+    ref
+  ) => {
+    const handleListItemSelect = () => {
+      if (content && handleOptionClick) {
+        handleOptionClick(content as never);
+      }
+    };
+
+    return (
+      <div
+        css={listItemStyle({ size, selected, highlighted, disabled })}
+        ref={selected ? ref : null}
```

Comment:
why you check this here? i think passing `ref={ref}` always could work as well

Reply:
This ref is for the selected Item. if I pass the ref to every ListItem then when the Select Dropdown opens, it will scroll to every Item, not on the selected one only. So without this check we have this: 
![selectRef](https://user-images.githubusercontent.com/57801353/127865150-c29361c8-36a8-409a-968d-d4cd17dda78c.gif)